### PR TITLE
test(artifacts): add regression test for empty variant playlist resolution

### PR DESCRIPTION
**Pull request by ManuGH**
Addresses the missing regression test from PR #683.

### CODE REVIEW

File: backend/internal/control/http/v3/recordings/artifacts/resolver_test.go [hunk 1]
```diff
@@ -334,4 +334,20 @@ func TestArtifactResolver_ResolvePlaylistState(t *testing.T) {
 		assert.Equal(t, []byte("#EXTM3U\n#EXT-X-PLAYLIST-TYPE:EVENT"), res.Data)
 		assert.Empty(t, mgr.ActiveJobIDs(), "state query should not trigger a build job")
 	})
+
+	t.Run("Empty Variant -> Resolves to Default Copy Profile", func(t *testing.T) {
+		// Construct the expected default hash
+		target := recordingTargetProfile("")
+		expectedVariant := target.Hash()
```
[the remark this answers is not in the document]
Comment (ManuGH) [reply backend/internal/control/http/v3/recordings/artifacts/resolver_test.go]: Fixed in 684ff98d — the test now mirrors the nil fallback from ResolvePlaylistState (default copy profile + CanonicalizeTarget before hashing).

File: backend/internal/control/http/v3/recordings/artifacts/resolver_test.go [hunk 2]
```diff
@@ -334,4 +334,20 @@ func TestArtifactResolver_ResolvePlaylistState(t *testing.T) {
...
+
+		dir, _ := v3recordings.RecordingVariantCacheDir(cfg.HLS.Root, ref, expectedVariant)
+		os.MkdirAll(dir, 0755)
+		os.WriteFile(filepath.Join(dir, "index.m3u8"), []byte("#EXTM3U"), 0644)
```
[the remark this answers is not in the document]
Comment (ManuGH) [reply backend/internal/control/http/v3/recordings/artifacts/resolver_test.go]: Fixed in 684ff98d — setup errors from RecordingVariantCacheDir/MkdirAll/WriteFile are now asserted with assert.NoError in both subtests.
